pallets/delegation/src/delegation_hierarchy.rs

```diff
@@ -52,87 +57,158 @@ impl Default for Permissions {
 	}
 }
 
-/// A node representing a delegation hierarchy root.
-#[derive(Clone, Debug, Encode, Decode, PartialEq)]
-pub struct DelegationRoot<T: Config> {
-	/// The hash of the CTYPE that delegated attesters within this trust
-	/// hierarchy can attest.
-	pub ctype_hash: CtypeHashOf<T>,
-	/// The identifier of the root owner.
-	pub owner: DelegatorIdOf<T>,
-	/// The flag indicating whether the root has been revoked or not.
-	pub revoked: bool,
-}
+// The old types.
+pub(crate) mod v0 {
```

Comment:
maybe we could move those definitions into their own file? Looks also better when there will be more versions

Reply:
Agree. Thanks, I also wanted to point this out. Could be `deprecated.rs` or something like this.

Reply:
Addressed in https://github.com/KILTprotocol/mashnet-node/commit/0040f2d75ca135d86b18c86cea6f54e0cb7e24e5.